src/Functions/FunctionsStringSimilarity.cpp

```diff
@@ -108,6 +108,10 @@ struct NgramDistanceImpl
 
         if constexpr (case_insensitive)
         {
+#if defined(MEMORY_SANITIZER)
+            /// Due to PODArray padding accessing more elements should be OK
```

Comment:
Speaking of tests/msan_suppressions.txt, it contains another suppression

```
# We apply std::tolower to uninitialized padding, but don't use the result, so        
# it is OK. Reproduce with "select ngramDistanceCaseInsensitive(materialize(''), '')" 
fun:tolower                                                                           
```

Looks like msan16 no longer picks these suppressions up.